### DISCUSSION TITLE
New environment

### DISCUSSION THREAD

**Discussion by w-disaster** [instadeepai/Mava#820] [General]
Hi there,
What's the best way to implement a completely new environment? Should I make use of the ```dm_env``` API?

Best,
Luca

**Comment by sash-a** [comment]
Hey @w-disaster I would say the best way is to look at some of our examples, we use a slightly modified version of the `dm_env` API. 

See the instructions in the readme in this folder and the examples in the same folder https://github.com/instadeepai/Mava/discussions/

**Comment by KaleabTessera** [comment]
Hi @w-disaster , we some instructions [here](https://github.com/instadeepai/Mava/blob/develop/mava/wrappers/README.md#getting-a-new-environment-to-work-with-mava). Are these clear enough or can we provide more information?
